x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/inventory_metric_threshold_executor.ts

```diff
@@ -175,13 +176,14 @@ export const createInventoryMetricThresholdExecutor = (libs: InfraBackendLibs) =
         );
 
       const compositeSize = libs.configuration.alerting.inventory_threshold.group_by_page_size;
+      const { dateEnd } = getTimeRange();
       const results = await Promise.all(
         criteria.map((condition) =>
           evaluateCondition({
             compositeSize,
             condition,
             esClient,
-            executionTimestamp: startedAt,
+            executionTimestamp: new Date(dateEnd),
```

Comment:
are we sure that we want to use `dateEnd` and not `dateStart`? I am pretty sure I am missing something here, can you explain me the logic?

Reply:
So the `executionTimestamp` is used [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/evaluate_condition.ts#L55) in the `to` field. Please let me know if I am missing something.